src/ray/util/event.h

```diff
@@ -37,10 +37,15 @@ using json = nlohmann::json;
 
 namespace ray {
 
-#define RAY_EVENT(event_type, label)                                \
-  if (ray::RayEvent::IsLevelEnabled(                                \
-          ::ray::rpc::Event_Severity::Event_Severity_##event_type)) \
-  ::ray::RayEvent(::ray::rpc::Event_Severity::Event_Severity_##event_type, label)
+#define RAY_EVENT(event_type, label)                                            \
+  if (ray::RayEvent::IsLevelEnabled(                                            \
+          ::ray::rpc::Event_Severity::Event_Severity_##event_type) ||           \
+      ray::RayLog::IsLevelEnabled(ray::RayEvent::EventLevelToLogLevel(          \
+          ::ray::rpc::Event_Severity::Event_Severity_##event_type)))            \
+  ::ray::RayEvent(::ray::rpc::Event_Severity::Event_Severity_##event_type,      \
+                  ray::RayEvent::EventLevelToLogLevel(                          \
+                      ::ray::rpc::Event_Severity::Event_Severity_##event_type), \
+                  label)
```

Comment:
should we use something like this?
```
#define RAY_EVENT(event_type, label)                                            \
  if (ray::RayEvent::IsLevelEnabled(                                            \
          ::ray::rpc::Event_Severity::Event_Severity_##event_type) ||           \
      ray::RayLog::IsLevelEnabled(ray::RayLog::event_type))                     \
  ::ray::RayEvent(::ray::rpc::Event_Severity::Event_Severity_##event_type,      \
                  label,  __FILE__, __LINE__)
```
So that the following code 
```
RAY_EVENT(ERROR, label)
```
will be expanded to
```
if (ray::RayEvent::IsLevelEnabled( ::ray::rpc::Event_Severity::Event_Severity_ERROR) || ray::RayLog::IsLevelEnabled(ray::RayLog::ERROR)) ::ray::RayEvent(::ray::rpc::Event_Severity::Event_Severity_ERROR, label, "/app/example.cpp", 8)
```


And you just need to implement a following RayEvent constructor:
```
RayEvent(rpc::Event_Severity severity, const std::string &label, const char* file, int64_t line)
```

https://godbolt.org/z/anbqjW8xj

Reply:
Good for me overall. But I can't use `ray::RayLog::IsLevelEnabled(ray::RayLog::event_type))` because I convert fatal events to error logs here https://github.com/ray-project/ray/pull/18376/files#diff-b46837ecea32982fc0315ea377a04bd8b3639226c0132cf0387d65e3ab76f548R252